server/mainapp/models.py

```diff
@@ -123,3 +123,183 @@ def fetch_data(self) -> response.JsonResponse:
             return docs
 
         raise DataFetchingError("There Are No Posts In The Database At This Moment")
+
+
+class Likes:
+    def __init__(self) -> None:
+        """
+        Connect to MongoDB
+        """
+        client = pymongo.MongoClient(DATABASE["mongo_uri"])
+        self.db = client[DATABASE["db"]]["Likes"]
+
+    def insert_data(
+            self,
+            music_data_pid: str,
+            user_email: str,
+    ) -> None:
+        """Insert file name and data into db
+
+        Args:
+            music_data_pid: FK for pid of the post
+            user_email: FK for user email
+
+        Returns:
+            None
+        """
+        if self.db.find_one({"music_data_pid": music_data_pid, "user_email": user_email}):
+            raise FileAlreadyExistsForCurrentUserError(
+                f"Like for post {music_data_pid} Already Exists For User {user_email}"
+            )
+
+        data = {
+            "music_data_pid": music_data_pid,
+            "user_email": user_email,
+        }
+        self.db.insert_one(data)
+
+    def delete_data(self, music_data_pid: str, user_email: str) -> None:
```

Comment:
It might be better to rename this function to "remove_like" - It is more apt for the functionality it is providing. Also again, keeping a track of the likes is important as we can update it on the front end